internal/nonkube/client/fs/custom_resource_handler.go

```diff
@@ -2,20 +2,27 @@ package fs
 
 import (
 	"fmt"
+	"io/fs"
 	"os"
 	"path/filepath"
+
 	"sigs.k8s.io/yaml"
 )
 
 type CustomResourceHandler[T any] interface {
 	Add(T) error
 	Update(T) error
-	Get(name string) T
+	Get(name string) (T, error)
+	GetRuntime(name string) (T, []fs.DirEntry, error)
```

Comment:
The idea of the CustomResourceHandler interface was to have a CRUD methods, similar to the kubernetes client, like: `Add`, `Update`, `Get` and `Delete`. If we need a method that gets more than one element we should create `List`. I think that we should have a `Get` method that retrieves a resource by its name, and a `List` methods that retrieves all the resources of the same type.
Also, this `List` method should return a list of Connectors, instead of DirEntry, the implementation of the method should take care of the transformation from file to connectors.

Reply:
The `List` method should replace `GetRuntime`

Reply:
done